src/test/librados/cls_remote_reads.cc

```diff
@@ -0,0 +1,52 @@
+#include <set>
+#include <string>
+
+#include "common/ceph_json.h"
+#include "gtest/gtest.h"
+#include "test/librados/test_cxx.h"
+
+using namespace librados;
+
+TEST(ClsTestRemoteReads, TestGather) {
```

Comment:
```
2021-03-25T20:39:37.253 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [ RUN      ] ClsTestRemoteReads.TestGather
2021-03-25T20:39:37.254 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-2494-g51f534a
d/rpm/el8/BUILD/ceph-17.0.0-2494-g51f534ad/src/test/librados/cls_remote_reads.cc:46: Failure
2021-03-25T20:39:37.254 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: Expected equality of these values:
2021-03-25T20:39:37.254 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads:   0
2021-03-25T20:39:37.254 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads:   ioctx.exec("tgt_object", "test_remote_reads", "test_gather", in, out)
2021-03-25T20:39:37.254 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads:     Which is: -1
2021-03-25T20:39:37.255 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [  FAILED  ] ClsTestRemoteReads.TestGather (7437 ms)
2021-03-25T20:39:37.255 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [----------] 1 test from ClsTestRemoteReads (7437 ms total)
2021-03-25T20:39:37.255 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads:
2021-03-25T20:39:37.255 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [----------] Global test environment tear-down
2021-03-25T20:39:37.255 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [==========] 1 test from 1 test suite ran. (7437 ms total)
2021-03-25T20:39:37.256 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [  PASSED  ] 0 tests.
2021-03-25T20:39:37.256 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [  FAILED  ] 1 test, listed below:
2021-03-25T20:39:37.256 INFO:tasks.workunit.client.0.smithi166.stdout:     api_cls_remote_reads: [  FAILED  ] ClsTestRemoteReads.TestGather
```

see https://pulpito.ceph.com/kchai-2021-03-25_14:56:28-rados-wip-kefu-testing-2021-03-25-2042-distro-basic-smithi/5998005/

Reply:
@kiizawa Looks like the new tests failed?  The above link should give you the relevant logs.

Reply:
@athanatos, @tchaikov 

I took a look at the OSD logs.
It seems like OSDs failed to load libcls_test_remote_reads.so due to a permission problem.

> 2021-03-25T20:39:37.195+0000 7f952c0be700 20 osd.5 pg_epoch: 68 pg[24.2( empty local-lis/les=67/68 n=0 ec=67/67 lis/c=67/67 les/c/f=68/68/0 sis=67) [5,4,0] r=0 lpr=67 crt=0'0 mlcod 0'0 active+clean] do_op: op osd_op(client.24703.0:5 24.2 24:41780a1c:::tgt_object:head [call test_remote_reads.test_gather in=237b] snapc 0=[] ondisk+read+known_if_redirected e68) v8
>
> 2021-03-25T20:39:37.195+0000 7f952c0be700  0 _get_class not permitted to load test_remote_reads

Would it be possible to run the test with the following options?

> [osd]
> osd class load list = *
> osd class default list = *

Alternatively, I could modify src/common/options.cc so that these lists include "test_remote_reads" by default.

Reply:
The rados_api_tests.yaml file can provide osd config overrides.  I think what you want is probably:

```
diff --git a/qa/suites/rados/basic/tasks/rados_api_tests.yaml b/qa/suites/rados/basic/tasks/rados_api_tests.yaml
index a54e03b1a9c..a1b136f7cc2 100644
--- a/qa/suites/rados/basic/tasks/rados_api_tests.yaml
+++ b/qa/suites/rados/basic/tasks/rados_api_tests.yaml
@@ -16,6 +16,9 @@ overrides:
         debug ms: 1
       mon:
         mon warn on pool no app: false
+      osd:
+        osd class load list: *
+        osd class default list: *
 tasks:
 - workunit:
     clients:
```

Reply:
I'm a bit sceptical about exposing this as a default class as its purpose is really for testing.

Reply:
@kiizawa ^

Reply:
Btw, there's a brand new trio of video guides on how to use/develop teuthology tests if you are interested.  https://youtu.be/3ZYBMGfQKVk is the first one.

Reply:
@athanatos

I modified the rados_api_tests.yaml file.

Thanks, I will watch the video.